### DISCUSSION TITLE
Choice of mode of use

### DISCUSSION THREAD

**Discussion by Bocelli** [seerge/g-helper#1904] [Ideas]
Hello everyone, I am an owner of Rog Ally and since I discovered G-Helper I have always and only used this program. As I used it, I noticed that armory crate occasionally got "in the way" for a few seconds. Tired of this, I decided to disable the Asus services, obviously leading to the loss of the overlay in the photo. Normally it doesn't bother me but the loss of the control mode selector is a step that I can hardly overcome (I often change from Desktop to Gaming and I don't keep it automatic because sometimes the Rog Ally doesn't recognize the game I'm playing) . I was wondering if it was possible to add a function that simulates the mode change in G-Helper (or if you know of any application that allows me to do this change so I can integrate it into the shortcuts of one of the keys)
![Senza titolo](https://github.com/seerge/g-helper/assets/57041651/15807c69-6d30-4744-b6b4-a1acd3e24f51)

**Comment by seerge** [comment]
@Bocelli hello, unfortunately I don't know how it's controlled atm :) I have a feeling that Armoury simply emulates mouse movements programmatically by constantly listening to controller. So it's not just ON/OFF switch.

**Reply by Bocelli** [reply]
It works exactly like this, with the right analog you move the cursor while with RB and RT you click the mouse. I would like it if it were possible, even through a third-party application, to have this ON and OFF but I assume nothing can be done about it

**Comment by seerge** [comment]
@Bocelli you can emulate mouse via controlled by using Steam for example :)

**Reply by Bocelli** [reply]
it works great!  thank you for the idea!

**Comment by seerge** [comment]
@Bocelli Actually, I had to dig a bit deeper and managed to find a way to toggle native Gamepad / Mouse modes for Ally.

Unlike Steam this toggle works on hardware level (so controlled itself starts working as a mouse)

I have also implemented experimental "Auto" mode, where G-Helper would toggle Gamepad if it detects a game where AMD Driver reports FPS counter (i.e. something graphical), in other cases it would toggle Mouse mode

Keep in mind, that to make sure this things works  - Armoury Stuff needs to be disabled, or they both will do same

Try this build
[GHelper.zip](https://github.com/seerge/g-helper/files/13928068/GHelper.zip)

**Reply by Bocelli** [reply]
Wow! I'll try it right away and let you know, thank you so much for your speed!

**Reply by Bocelli** [reply]
I've tested this version of G-Helper a bit and let's say it works quite well.  Like you said, everything that can be added to AMD Adrenaline works flawlessly.  I obviously find it difficult to make everything work with the Game Pass games but just make a switch to the GamePad and everything is resolved

**Reply by Bocelli** [reply]
The only advice, if I may, is to add the mode change as a shortcut

**Reply by seerge** [reply]
@Bocelli it's not about AMD Adrenaline, it's pretty much any app / game that is seen by driver with non zero FPS.

I tested on Forza /  Tomb Raider from Gamepass and it detects them as well :)

**Reply by Bocelli** [reply]
@seerge I tried Sea of Stars and it still worked with the cursor.  I'll try again then :) thank you for your effort, I know you're not very interested in working for Rog Ally so it's more of an inconvenience than anything else

**Reply by seerge** [reply]
@Bocelli I just tried Sea of Stars, and it also detects game correctly (you can check app log for a lines like ) 
```
1/13/2024 11:29:52 PM: ControllerAuto Mouse Feature 1ABE|64: 5A-D1-01-01-03
1/13/2024 11:29:52 PM: -1
1/13/2024 11:32:33 PM: ControllerAuto Gamepad Feature 1ABE|64: 5A-D1-01-01-01
1/13/2024 11:32:33 PM: 119.83224
1/13/2024 11:33:04 PM: ControllerAuto Mouse Feature 1ABE|64: 5A-D1-01-01-03
1/13/2024 11:33:04 PM: -1
```

First is action (what mode g-helper sets), second is measured FPS (i.e. if it.s -1, it's not a game)

I did few more updates and also added a fancy low-level backlight button and FPS limiter to the app purely for the Ally
Try this build

[GHelper.zip](https://github.com/seerge/g-helper/files/13929735/GHelper.zip)

**Reply by seerge** [reply]
P.S. I would recommend to keep Radeon Software on startup (it's part of amd "driver" set, and is responsible for amd's FPS measurement and in-game overlay). If it's not started - it could be a reason why G-Helper doesn't get FPS readings and don't see it as a game

![Screenshot 2024-01-13 233622](https://github.com/seerge/g-helper/assets/5920850/72576882-c73a-48b3-b628-b6c1edebf474)

**Reply by Bocelli** [reply]
@seerge You are right! I tried again on Sea of Stars and it works great. It was almost certainly Playnite's fault as it conflicted with something (unfortunately having a decent launch interface is impossible without buying SteamDeck or relying on some application like Playnite). I tried the latest version you uploaded and I like and appreciate the work you did.

However, since we are "in the running", I wanted to ask you: could you add the mode change as a shortcut in the "extra" item? So if I have a problematic game I don't necessarily have to minimize it to change the control mode.

Another question (sorry if there are so many but G-Helper is the only program I'm so confident with): I saw that both Universa X86 and Companion Handheld have the toggle for adaptive TDP. Could you replicate the same thing? Because Rog Ally in one way or another almost always tends to consume the wattage set by the profile even if the game requires less

**Reply by seerge** [reply]
@Bocelli 

I can add a binding to Extra, but imho since you can bind only 3 buttons, it's a bit of a waste. That's why I have added whole Auto thing. 

As for auto TDP - i'm not really sure if it's even worth it. On my experience just limiting FPS (via radeon settings that is now also available in g-helper built I posted here), has almost same effect. Hardware won't use more than it needs to.

**Reply by Bocelli** [reply]
@seerge Consider that the situation of my shortcuts is currently this.  But if it's not possible I won't say anything else, you've already done enough at least personally
![IMG_20240114_122249_051](https://github.com/seerge/g-helper/assets/57041651/efa115e5-ccb3-42c6-97a3-d37d8644196d)

**Reply by Bocelli** [reply]
@seerge HI!  I'm trying to play on Yuzu but the mode switch doesn't seem to work.  I tried adding the exe file to AMD Adrenaline but the situation does not change

[Video here](https://youtu.be/Krgfs98Fzb0?feature=shared)

**Reply by seerge** [reply]
@Bocelli what I have noticed, when Ally wakes up from sleep, controller sometimes refuses to get any commands (i.e it keep sleeping)

That seem to be a reason on your video

Latest https://github.com/seerge/g-helper/releases/download/v0.148/GHelper.zip runs a wake up command, every time it sets a mode. Try it instead :)
